components/script/dom/document.rs

```diff
@@ -2261,19 +2265,12 @@ impl DocumentMethods for Document {
 
     // https://html.spec.whatwg.org/multipage/#dom-document-hasfocus
     fn HasFocus(&self) -> bool {
-        match self.browsing_context() {
-            Some(browsing_context) => {
-                // Step 2.
-                let candidate = browsing_context.active_document();
-                // Step 3.
-                if &*candidate == self {
-                    true
-                } else {
-                    false //TODO  Step 4.
-                }
-            }
-            None => false,
+        // Step 1-2.
+        if self.window().parent_info().is_none() && self.is_fully_active() {
+            return true;
         }
+        // TODO Step 3.
```

Comment:
I think we should find a way to do this, and other cases that may have regressed with this PR before merging.